iterator_practice.cpp

```diff
@@ -1,16 +1,13 @@
 #include <iostream>
-#include <iterator> // for iterator
```

Comment:
I don't think you need this after my suggested changes...you only shoukd include it if you use it yourself, but not if you just use something that uses it, like vector. The principle is to "include what you use."